bbc/simorgh
# Lite page rendering

Overall changes
======
- Adds 'Lite' page rendering to the application via the `.lite` url extension, for example: http://localhost:7080/pidgin.lite?renderer_env=live
- This is purely server-side rendered with no JS shipped to the client (meaning no hydration in React)
- JS can be added using the `<Helmet>` component to inject into the `<head>`. You may see the Comscore `main-1.0.js` file in the Chrome 'Network' tab from this
- Exposes `isLite` flag that can be accessed from the `RequestContext` context to add conditional logic for the Lite renderer
- Exposes `isLite` flag that can be accessed in the `css()` via the theme to allow direct styling when the app is in 'Lite' mode
- Separates out our rendering platforms in `document/component` to make it more clearer/easier to maintain
- Adds `.lite` route regex to all page types
- Updates Next.js `_document` to expose Emotion CSS `css` and `ids` to allow them to be used correctly in the 'Lite' renderer
- Adds `Lite Mode` toggle to Storybook **(Note: this may require per-story amendments to allow the story to respect the `isLite` flag in the `RequestContext`. Check the 'ArticlePage' story amended in this PR, whereby the `RequestContext` in the story was removed as it was unneeded and overriding the global `RequestContext` in the `preview.tsx` Storybook file)**

Testing
======
1. Visit http://localhost:7080/pidgin.lite?renderer_env=live
2. Confirm the page loads in the new Lite renderer
3. Visit http://localhost:7080/pidgin/articles/cd14lllvdveo.lite?renderer_env=live
4. Confirm the pages loads in the new Lite renderer
5. Visit http://localhost:7081/pidgin/live/c7p765ynk9qt.lite?renderer_env=test in the Next.js app
6. Confirm the page loads in the new Lite renderer
*Note*: You will be able to tell its in the new Lite renderer by checking the 'Network' tab in Chrome and confirming that no JS is loaded

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
